python/cudf/groupby/groupby.py

```diff
@@ -108,6 +142,9 @@ def _apply_agg(self, agg_type, result, add_col_values,
             if agg_type == "count":
                 out_col_agg_series = Series(
                     Buffer(rmm.device_array(col_agg.size, dtype=np.int64)))
+            elif agg_type == "mean":
+                out_col_agg_series = Series(
+                    Buffer(rmm.device_array(col_agg.size, dtype=np.float64)))
```

Comment:
This isn't true for all cases though, right? Unsure if we can match Pandas behavior perfectly here given it's data-specific...

Reply:
My current fix doesn't produce broken results, but it does produce inconsistent results. Integer truncation of a mean function is probably bad.

I agree that a general refactor is incoming on this component.  There's lots of other small changes that I'm working through to inform me of strategy toward the general refactor.  It is going to get a little messier still before I can refactor it, but I am optimistic that I'll be able to handle many of @mrocklin's cases in a short time period.

Reply:
Presently we can't trivially match Pandas - in the current structure we'd need to do a full groupby on the data with a "native-dtype" column and another with a `float64` column and compare them to figure out the correct column. ..